src/Utilities/Compiler/Extensions/OperationBlockAnalysisContextExtension.cs

```diff
@@ -63,6 +64,19 @@ static bool IsSingleStatementBody(IBlockOperation body)
 
             return false;
         }
+
+        /// <summary>
+        /// Auto-properties (readonly, get, set) marked with an attribute have a block context callback while they don't if
+        /// there is no attribute (see https://github.com/dotnet/roslyn/issues/46132).
+        /// Auto-properties only have 'OperationKind.None' operations.
+        /// </summary>
+#pragma warning disable RS1012 // Start action has no registered actions.
+        public static bool IsAutoProperty(this OperationBlockStartAnalysisContext context)
```

Comment:
Lets change this to completely symbol based logic as discussed in https://github.com/dotnet/roslyn/issues/46682. Given we will not have access to access to new API added in https://github.com/dotnet/roslyn/commit/83ecbec0733d65f892f48c090120a923512b4652, we can look for all the field symbols on the property's containing type and check if any field's associated symbol is the property symbol.

```csharp
public static bool IsAutoPropertyAccessor(this IMethodSymbol method)
    => methodSymbol.IsPropertyAccessor() && methodSymbol.AssociatedSymbol is IPropertySymbol propertySymbol && methodSymbol.ContainingType.GetMembers().OfType<IFieldSymbol>.Any(f => propertySymbol.Equals(f.AssociatedSymbol);
```

Reply:
I feel this would be much more stable logic then relying on operation tree shape and root operation node kinds. Please make sure to move this method over to `IMethodSymbolExtensions`.

Reply:
Thank you @mavasani that's indeed a lot clearer and a more robust approach.